packages/recorder/src/main.tsx

```diff
@@ -26,22 +26,31 @@ export const Main: React.FC = ({
   const [log, setLog] = React.useState(new Map<string, CallLog>());
   const [mode, setMode] = React.useState<Mode>('none');
 
-  window.playwrightSetMode = setMode;
-  window.playwrightSetSources = React.useCallback((sources: Source[]) => {
-    setSources(sources);
-    window.playwrightSourcesEchoForTest = sources;
+  React.useLayoutEffect(() => {
```

Comment:
why blocking effect?

Reply:
This is the way these side effects are supposed to be triggered in React (though ideally it would be `useEffect`, but I figured people would complain about it potentially being delayed too long). Prior to this change, they were being set on each render of the main component (which happens quite often since it receives the data from the bridge). Unnecessarily inefficient, and for no gain.